PHPCI/Plugin/Stash.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * PHPCI - Continuous Integration for PHP
+ *
+ * @copyright    Copyright 2014, Block 8 Limited.
+ * @license      https://github.com/Block8/PHPCI/blob/master/LICENSE.md
+ * @link         https://www.phptesting.org/
+ */
+
+namespace PHPCI\Plugin;
+
+use b8\View;
+use PHPCI\Builder;
+use PHPCI\Helper\Lang;
+use PHPCI\Model\Build;
+
+/**
+* Stash Plugin - Provides simple stash capability to PHPCI.
+* @author       Alasdair Campbell <alasdair@campbelldiagnostics.co.uk>
+* @package      PHPCI
+* @subpackage   Plugins
+*/
+class Stash implements \PHPCI\Plugin
+{
+    /**
+     * @var \PHPCI\Builder
+     */
+    protected $phpci;
+
+    /**
+     * @var \PHPCI\Model\Build
+     */
+    protected $build;
+
+    /**
+     * @var array
+     */
+    protected $options;
+
+    /**
+     * @var string
+     */
+    protected $phpciHostname = null;
+
+    /**
+     * @var string
+     */
+    protected $stashHostname = null;
+
+    /**
+     * @var integer
+     */
+    protected $stashPort = 7990;
+
+    /**
+     * @var string
+     */
+    protected $username = null;
+
+    /**
+     * @var string
+     */
+    protected $password = null;
+
+    /**
+     * Set up the plugin, configure options, etc.
+     * @param Builder $phpci
+     * @param Build $build
+     * @param array $options
+     */
+    public function __construct(
+        Builder $phpci,
+        Build $build,
+        array $options = array()
+    ) {
+        $this->phpci        = $phpci;
+        $this->build        = $build;
+        $this->options      = $options;
+        
+        $this->setProperties();
```

Comment:
I prefer to pass ```$options``` to ```setProperties``` and remove the ```$this->options``` (since you are not using it), or (imho more consistent with other plugins) use ```$this->options['stash_name']``` without ```$this->stashName``` ....